runtime/compiler/p/env/J9CPU.cpp

```diff
@@ -135,21 +47,10 @@ J9::Power::CPU::isCompatible(const OMRProcessorDesc& processorDescription)
    return targetProcessor == processor;
    }
 
-OMRProcessorDesc
-J9::Power::CPU::getProcessorDescription()
-   {
-#if defined(J9VM_OPT_JITSERVER)
-   if (auto stream = TR::CompilationInfo::getStream())
-      {
-      auto *vmInfo = TR::compInfoPT->getClientData()->getOrCacheVMInfo(stream);
-      return vmInfo->_processorDescription;
-      }
-#endif /* defined(J9VM_OPT_JITSERVER) */
-   return _processorDescription;
-   }
-
-bool
-J9::Power::CPU::getPPCSupportsVSX()
-   {
-   return self()->supportsFeature(OMR_FEATURE_PPC_HAS_VSX);
-   }
+void
+J9::Power::CPU::setProcessorType(OMRProcessorArchitecture p)
```

Comment:
does this routine exist only temporarily?  I am imagining that these p/env/J9CPU.x files would be deleted completely once everything CPU-related is moved to OMR side.  it is less ideal that CPU-related stuffs are still split implementation.

Reply:
I tried to remove as much as possible but this function can't be removed due to JITServer. 
```cpp
OMRProcessorDesc
J9::Power::CPU::getProcessorDescription()
   {
#if defined(J9VM_OPT_JITSERVER)
   if (auto stream = TR::CompilationInfo::getStream())
      {
      auto *vmInfo = TR::compInfoPT->getClientData()->getOrCacheVMInfo(stream);
      return vmInfo->_processorDescription;
      }
#endif /* defined(J9VM_OPT_JITSERVER) */
   return _processorDescription;
   }
```

for `J9::Power::CPU::setProcessorType` I would say it only exists temporarily. It's only here to make this function `TR::CompilationInfo::setProcessorByDebugOption()` happy for now. And eventually I will sink it down to OMRCPU. Is this function still being used? Also I have a question regarding `setProcessorByDebugOption()`, notice how it only changes the processor type and doesn't change the processor features, would that be a problem?

Reply:
> Also I have a question regarding setProcessorByDebugOption(), notice how it only changes the processor type and doesn't change the processor features, would that be a problem?

it is a problem in theory, since features can be disabled individually on running instance of OS.  But, i would say it is not a problem usually. Running instances of OS (same level) typically have the same set of features enabled. Especially for Linux (targeted containerized environment on POWER), it is not that easy to disable a hardware feature (without rebuilding the kernel).  AIX can enable/disable more dynamically through management tool, but it doesn't support Docker anytime soon.